x-pack/plugins/alerting/server/alerts_client/alerts_client.test.ts

```diff
@@ -1666,6 +1674,229 @@ describe('Alerts Client', () => {
+            )
+          ).rejects.toThrowError(
+            'Must specify rule ID, space ID, and executionUuid for scoped query AAD alert query.'
+          );
+        });
+
+        test('should throw if spaceId is not specified', async () => {
+          const alertsClient = new AlertsClient(alertsClientParams);
+          const { spaceId, ...paramsWithoutRuleId } = getParamsByMaintenanceWindowScopedQuery;
+
+          await expect(
+            alertsClient.getMaintenanceWindowScopedQueryAlerts(
+              paramsWithoutRuleId as GetMaintenanceWindowScopedQueryAlertsParams
+            )
+          ).rejects.toThrowError(
+            'Must specify rule ID, space ID, and executionUuid for scoped query AAD alert query.'
+          );
+        });
+
+        test('should throw if executionUuid is not specified', async () => {
+          const alertsClient = new AlertsClient(alertsClientParams);
+          const { executionUuid, ...paramsWithoutRuleId } = getParamsByMaintenanceWindowScopedQuery;
+
+          await expect(
+            alertsClient.getMaintenanceWindowScopedQueryAlerts(
+              paramsWithoutRuleId as GetMaintenanceWindowScopedQueryAlertsParams
+            )
+          ).rejects.toThrowError(
+            'Must specify rule ID, space ID, and executionUuid for scoped query AAD alert query.'
+          );
+        });
+      });
+
+      describe('updateAlertMaintenanceWindowIds', () => {
+        test('should update alerts with new maintenance window Ids', async () => {
+          const alertsClient = new AlertsClient(alertsClientParams);
+
+          const alert1 = new Alert('1', { meta: { maintenanceWindowIds: ['mw1'] } });
+          const alert2 = new Alert('2', { meta: { maintenanceWindowIds: ['mw1', 'mw2'] } });
+          const alert3 = new Alert('3', { meta: { maintenanceWindowIds: ['mw2', 'mw3'] } });
+
+          jest.spyOn(LegacyAlertsClient.prototype, 'getProcessedAlerts').mockReturnValueOnce({
+            '1': alert1,
+            '2': alert2,
+            '3': alert3,
+          });
+
+          await alertsClient.updateAlertMaintenanceWindowIds([
+            alert1.getUuid(),
+            alert2.getUuid(),
+            alert3.getUuid(),
+          ]);
+
+          const params = clusterClient.updateByQuery.mock.calls[0][0] as UpdateByQueryRequest;
+
+          expect(params.query).toEqual({
+            terms: {
+              _id: [alert1.getUuid(), alert2.getUuid(), alert3.getUuid()],
+            },
+          });
+
+          expect(params.script).toEqual({
+            source: expect.anything(),
+            lang: 'painless',
+            params: {
+              [alert1.getUuid()]: ['mw1'],
+              [alert2.getUuid()]: ['mw1', 'mw2'],
+              [alert3.getUuid()]: ['mw2', 'mw3'],
+            },
+          });
+        });
+      });
+
+      describe('updateAlertsMaintenanceWindowIdByScopedQuery', () => {
```

Comment:
I added a test to the inner function since we never await on `updateAlertsMaintenanceWindowIdByScopedQuery` so it's a bit hard to test the reject case.